clients/cobol-lsp-vscode-extension/src/services/copybook/CopybookURI.ts

```diff
@@ -87,19 +86,18 @@ export class CopybookURI {
     profileName: string,
     dataset: string,
   ): string {
-    const rootPath = vscode.workspace.workspaceFolders![0].uri.fsPath;
     return path.join(
-      rootPath,
-      C4Z_FOLDER,
+      Utils.getExtensionFolder(),
+      ZOWE_FOLDER,
       COPYBOOKS_FOLDER,
       profileName,
       dataset,
     );
   }
   /**
    * This method produce an array with element that following the schema
-   * "file://[WORKSPACE_FOLDER]/.c4z/.copybooks/PROFILE/DATASET" or
-   * "file://[WORKSPACE_FOLDER]/.c4z/.copybooks/PROFILE/USS"
+   * "file://[EXTENSION_FOLDER]/.zowe/.copybooks/PROFILE/DATASET" or
+   * "file://[EXTENSION_FOLDER]/.zowe/.copybooks/PROFILE/USS"
    * @param profile represent a name of a folder within the .copybooks folder that have the same name as the
    * connection name needed to download copybooks from mainframe.
    */
```

Comment:
Wouldn't it be better to just pass in the extension directory as a parameter?

Reply:
This will lead to refactoring in following as well considering download/access folders of copybooks are in responsibility of CopybookUri class : https://github.com/hizir-bc/che-che4z-lsp-for-cobol/blob/feature/swap-workspace-home/clients/cobol-lsp-vscode-extension/src/services/copybook/CopybookURI.ts#L82

What benefits do you see we can use of ?

Reply:
The benefit is, that there will be no need to keep a global state variable, and testing will be much simpler. Also all the other "services" can take the value as parameter on construction, again making testing easier.

Reply:
resolved at : 19c31007cf0f1c59dafd17843eeb8d6bae0c4e67